doc/build-instructions/Build_Instructions_V11.md

```diff
@@ -786,3 +787,351 @@ JCL      - fd5cc89 based on jdk-11.0.2+7)
   - *This product includes cryptographic software written by Eric Young (eay@cryptsoft.com).*
 
 :penguin: *Congratulations!* :tada:
+
+----------------------------------
+
+## Riscv64
+:rocket:
+The following instructions guide you through the process of building an **OpenJDK V11** binary that contains Eclipse OpenJ9 for Riscv64 (RISC-V 64-bit) Linux.
+
+:bulb:
+It is assumed that Ubuntu 16.04 is the default host system for the cross-compilation.
+
+### 1. Prepare your system for cross-compilation
+:rocket:
+A number of software packages/dependencies must be installed on the host system to create a suitable cross-compiling environment:
+
+- [The GNU cross-toolchain](https://github.com/riscv/riscv-gnu-toolchain), which contains the source code of the C/C++ cross-compiler intended for RISC-V.
+- [bootJDK](https://adoptopenjdk.net/archive.html?variant=openjdk11&jvmVariant=hotspot), which is only used in the cross-compilation on the host system.
+- [QEMU](https://www.qemu.org/download/#linux), which is an open source emulator that converts the RISC-V instructions to the opcode on the host system.
+- [Fedora_RISCV](https://fedorapeople.org/groups/risc-v/disk-images/), which contains an Fedora OS image and the corresponding BBL (Berkeley Boot Loader) binary file (used to boot via QEMU). 
+platform
+- [Freemarker V2.3.8](https://sourceforge.net/projects/freemarker/files/freemarker/2.3.8/freemarker-2.3.8.tar.gz/download)
+
+The following sections cover the whole installation & building steps for all required software packages.
+
+### 2. Install software packages for the cross-compilation
+:rocket:
+For the cross-compilation, the first thing is to install all required software packages on your system before building the JDK. When it comes to Fedora, you need to boot the OS image via QEMU to login the target system for installation.
+
+[1] First of all, download and install QEMU on your system as follows:
+```
+sudo apt-get install qemu
+```
+
+:bulb:
+If you are interested in building the QEMU source on your own, please refer to the building instructions
+at https://github.com/riscv/riscv-qemu/wiki for details.
+e.g.
+```
+git clone --recursive https://github.com/riscv/riscv-qemu.git
+cd riscv-qemu
+./configure \
+    --target-list=riscv64-softmmu,riscv64-linux-user
+make -j$(nproc)
+make install
+```
+
+[2] Download the Fedora OS image `stage4-disk.img.xz` and `bbl` (the boot loader) from
+https://fedorapeople.org/groups/risc-v/disk-images/ (4GB in size by default)
+and expend it appropriately (e.g. 20GB) for more space as follows:
+```
+unxz --keep stage4-disk.img.xz
+truncate -s 20G stage4-disk.img
+e2fsck -fp stage4-disk.img
+resize2fs stage4-disk.img
+```
+:bulb:
+Please follow the steps below to install `e2fsck` If it doesn't exist on your host system:
```

Comment:
This step should be done earlier: the previous step uses `e2fsck`.

Reply:
Adjusted the order of steps as suggested.